crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentCompletionService.java

```diff
@@ -0,0 +1,81 @@
+import uk.gov.justice.laa.crime.meansassessment.dto.maatcourtdata.DateCompletionRequestDTO;
+import uk.gov.justice.laa.crime.meansassessment.dto.maatcourtdata.FinancialAssessmentDTO;
+import uk.gov.justice.laa.crime.meansassessment.staticdata.enums.*;
+
+import java.time.LocalDateTime;
+
+import static java.util.Optional.ofNullable;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AssessmentCompletionService {
+
+    private final MaatCourtDataService maatCourtDataService;
+
+    public void execute(MeansAssessmentDTO assessment, String laaTransactionId) {
+        boolean isUpdateRequired = false;
+        MeansAssessmentRequestDTO meansAssessmentRequest = assessment.getMeansAssessment();
+        CurrentStatus assessmentStatus = meansAssessmentRequest.getAssessmentStatus();
+        AssessmentType type = meansAssessmentRequest.getAssessmentType();
+        if (CurrentStatus.COMPLETE.equals(assessmentStatus)) {
+            if (AssessmentType.INIT.equals(type)) {
+                isUpdateRequired = isInitAssessmentComplete(assessment);
+            } else {
+                isUpdateRequired = isFullUpdateRequired(assessment, laaTransactionId);
+            }
+        }
+
+        if (isUpdateRequired) {
+            updateApplicationCompletionDate(assessment, laaTransactionId);
+        }
+    }
+
+    void updateApplicationCompletionDate(MeansAssessmentDTO assessment, String laaTransactionId) {
+        assessment.setDateCompleted(LocalDateTime.now());
+        DateCompletionRequestDTO dateCompletionRequestDTO = DateCompletionRequestDTO
+                .builder()
+                .repId(assessment.getMeansAssessment().getRepId())
+                .assessmentDateCompleted(assessment.getDateCompleted())
+                .build();
+        maatCourtDataService.updateCompletionDate(dateCompletionRequestDTO, laaTransactionId);
+    }
+
+    boolean isFullUpdateRequired(MeansAssessmentDTO assessment, String laaTransactionId) {
+        Integer financialAssessmentId = ofNullable(
+                assessment.getMeansAssessment().getFinancialAssessmentId()
+        ).orElse(null);
+
+        if (financialAssessmentId != null) {
+            FinancialAssessmentDTO existingAssessment = maatCourtDataService.getFinancialAssessment(
+                    financialAssessmentId, laaTransactionId
+            );
+            return existingAssessment.getDateCompleted() == null;
+        }
+        return true;
+    }
+
+    boolean isInitAssessmentComplete(MeansAssessmentDTO assessment) {
+        InitAssessmentResult initResult = assessment.getInitAssessmentResult();
+        if (InitAssessmentResult.PASS.equals(initResult)) {
+            return true;
+        } else if (InitAssessmentResult.FAIL.equals(initResult)) {
+            CaseType caseType = assessment.getMeansAssessment().getCaseType();
+            MagCourtOutcome magCourtOutcome = assessment.getMeansAssessment().getMagCourtOutcome();
```

Comment:
This can be moved inside the else loop, or even we dont need to create this magCourOutcome variable as its only used once in the comparison.